nifi-nar-bundles/nifi-iotdb-bundle/nifi-iotdb-processors/src/main/java/org/apache/nifi/processors/PutIoTDBRecord.java

```diff
@@ -0,0 +1,283 @@
+                        tablet.addValue(id, rowIndex, value);
+                    }
+                    isFulled = tablet.rowSize == tablet.getMaxRowNumber();
+                }
+                if (isFulled) {
+                    if (aligned) {
+                        session.get().insertAlignedTablets(tablets);
+                    } else {
+                        session.get().insertTablets(tablets);
+                    }
+                    tablets.values().forEach(tablet -> tablet.reset());
+                }
+            }
+
+            AtomicBoolean hasRest = new AtomicBoolean(false);
+            tablets.forEach(
+                    (device, tablet) -> {
+                        if (hasRest.get() == false && tablet.rowSize != 0) {
+                            hasRest.set(true);
+                        }
+                    });
+            if (hasRest.get()) {
+                if (aligned) {
+                    session.get().insertAlignedTablets(tablets);
+                } else {
+                    session.get().insertTablets(tablets);
+                }
+            }
+        } catch (Exception e) {
+            getLogger().error("Processing failed {}", flowFile, e);
+            processSession.transfer(flowFile, REL_FAILURE);
+            e.printStackTrace();
+            return;
+        }
+        processSession.transfer(flowFile, REL_SUCCESS);
+    }
+
+    private IoTDBSchema getSchema(String property, RecordReader recordReader) throws Exception {
+        ValidationResult result;
+        IoTDBSchema schema;
+        result =
+                property != null
+                        ? validateSchemaAttribute(property)
+                        : validateSchema(recordReader.getSchema());
+
+        if (!result.getKey()) {
+            getLogger().error("The property `schema` has an error: {}", result.getValue());
+            throw new Exception();
+
+        } else {
+            if (result.getValue() != null) {
+                getLogger().warn("The property `schema` has a warn: {}", result.getValue());
+            }
+        }
+
+        schema =
+                property != null
+                        ? mapper.readValue(property, IoTDBSchema.class)
+                        : convertSchema(recordReader.getSchema());
+        return schema;
+    }
+
+    private long getTimestamp(IoTDBSchema schema, Record record) {
+        long timestamp;
+        Object time = record.getValue(schema.getTimeName());
+        if (time instanceof  Timestamp) {
+            Timestamp temp = (Timestamp) time;
+            timestamp = temp.getTime();
+        } else if (time instanceof  Time) {
+            Time temp = (Time) time;
+            timestamp = temp.getTime();
+        } else if (time instanceof  Date) {
+            Date temp = (Date) time;
+            timestamp = temp.getTime();
+        } else
+            timestamp = (Long) time;
+        return timestamp;
```

Comment:
It seems unsafe to default to `Long` without checking the type.
```suggestion
        } else if (time intanceof Long) {
            timestamp = (Long) time;
        } else {
            throw new IllegalArgumentException(String.format("Unexpected Time Field: %s", time));
        }
        return timestamp;
```